vickyctl/src/locks.rs

```diff
@@ -1,30 +1,433 @@
+            ctx.vicky_url,
+            lock_to_clear.id()
+        ))
+        .build()?;
+    client.execute(request)?.error_for_status()?;
+    Ok(())
+}
+
+fn handle_popup(
+    selected_task: &mut Option<usize>,
+    selected_button: &mut bool,
+    key: &KeyEvent,
+    args: &ResolveArgs,
+    locks: &mut Vec<PoisonedLock>,
+) -> Result<(), Error> {
+    if key.code == KeyCode::Left || key.code == KeyCode::Char('y') {
+        *selected_button = true;
+    } else if key.code == KeyCode::Right || key.code == KeyCode::Char('n') {
+        *selected_button = false;
+    }
+
+    if key.code == KeyCode::Char('y') || (key.code == KeyCode::Enter && *selected_button) {
+        unlock_and_refresh(args, locks, selected_task)?;
+    } else if key.code == KeyCode::Char('n') || (key.code == KeyCode::Enter && !*selected_button) {
+        *selected_task = None;
+    }
+
+    Ok(())
+}
+
+fn handle_events(
+    state: &mut TableState,
+    lock_amount: usize,
+    selected_task: &mut Option<usize>,
+    selected_button: &mut bool,
+    args: &ResolveArgs,
+    locks: &mut Vec<PoisonedLock>,
+) -> Result<bool, Error> {
+    if event::poll(std::time::Duration::from_millis(50))? {
+        if let Event::Key(key) = event::read()? {
+            if key.kind == event::KeyEventKind::Press {
+                if key.code == KeyCode::Char('q') || key.code == KeyCode::Esc {
+                    if selected_task.is_some() {
+                        *selected_task = None;
+                        return Ok(false);
+                    }
+                    return Ok(true);
+                }
+
+                match selected_task {
+                    None => handle_task_list(state, lock_amount, selected_task, &key),
+                    Some(_) => handle_popup(selected_task, selected_button, &key, args, locks)?,
+                }
+            }
+        }
+    }
+    Ok(false)
+}
+
+fn handle_task_list(
+    state: &mut TableState,
+    lock_amount: usize,
+    selected_task: &mut Option<usize>,
+    key: &KeyEvent,
+) {
+    match state.selected_mut() {
+        None => (),
+        Some(cur) => {
+            if (key.code == KeyCode::Up || key.code == KeyCode::Char('k')) && *cur > 0 {
+                *cur -= 1;
+            } else if (key.code == KeyCode::Down || key.code == KeyCode::Char('j'))
+                && *cur < lock_amount - 1
+            {
+                *cur += 1;
+            } else if key.code == KeyCode::Enter {
+                *selected_task = Some(*cur);
+            }
+        }
+    };
+}
+
+#[allow(dead_code)]
+fn get_longest_len<'a, T>(str_iter: T) -> u16
+where
+    T: Iterator<Item = &'a str>,
+{
+    str_iter
+        .map(|l| l.len())
+        .max()
+        .map_or(0, |len| u16::try_from(len).unwrap_or(u16::MAX))
+}
+
+fn draw_task_picker(f: &mut Frame, locks: &[PoisonedLock], state: &mut TableState) {
+    let rows: Vec<Row> = locks.iter().map(|l| l.into()).collect();
+
+    // let _widths: [Constraint; 4] = [
```

Comment:
Remove this, if it is not needed.

Reply:
I just wanted you to remove the commented out lines, not the feature itself 😂😂

Reply:
I was talking about the commented lines. They are a different view style of the feature. They aren't just "old code".